### ISSUE TITLE
Make annotation download behaviour consistent

### ISSUE THREAD

**Issue by MichaelBuessemeyer** [scalableminds/webknossos#6125]
## Context
While trying to automatically process an annotation for the merger mode, I stumbled upon the problem, that the download format for an annotation does not seem to be consistent. When a skeleton-only tracing is downloaded, the backend responds solely with the nml. If a volume annotation is downloaded the backend answers with a zip file containing the volume annotation data also in a zip for each data layer. If the volume annotation Is empty, the zip of it is also empty (which leads to bugs in wklibs. (see below). 

## Expected Behavior

It would be nice if the skeleton only tracing download would also have the nml wrapped in a zip file.
And it would be more consistent if the empty volume annotations would have a zip that is not empty but still has at least a single header.wkw file to tell that this layer / annotation is empty. 
## Current Behavior

As explained above: 
- The download of skeleton only tracings results in a plain nml. 
![image](https://user-images.githubusercontent.com/39529669/159760295-3d0ac349-4f9a-4fba-896b-e1c2847f11bd.png)
- The download of a hybrid tracing (and afaik volume only tracing) results in a zip containing the nml for the skeleton and the volume annotation layers in further zip files. If a volume layer is empty, the zip of it is also empty leading to unexpected behaviour in wklibs.
![image](https://user-images.githubusercontent.com/39529669/159759779-28259fd3-6e29-4ff6-9ede-f407343ab064.png)

## Steps to Reproduce the bug

- [ ] Cannot reproduce the bug anymore / needs deeper investigation.

1.  The following code using wklibs should fail right now with skeleton only
```python
with webknossos_context(WEBKNOSSOS_URL, user_auth_token):
        annotation = Annotation.download(
            f"{WEBKNOSSOS_URL}/annotations/{annotation_type}/{annotation_id}"
        ).skeleton
```
2. The following code using wklibs should fail right now with a hybrid tracing that has an empty annotation volume layer because nothing has been brushed in this annotation before.
```python
   temporary_annotation_path = (
            DATASETS_ROOT
            / organization_name
            / CONVERTING_DIR
            / run_id
            / "downloaded_annotation.zip"
    )
    temporary_annotation_path.parent.mkdir(exist_ok=True)
    with webknossos_context(WEBKNOSSOS_URL, user_auth_token):
        logger.info("Downloading input annotation...")
        annotation = Annotation.download(
            f"{WEBKNOSSOS_URL}/annotations/{annotation_type}/{annotation_id}"
        )
        annotation.save(temporary_annotation_path)
    with Annotation(
        annotation_zip_path
    ).temporary_volume_annotation_layer_copy() as input_annotation_layer:
        # cannot compute further as temporary_volume_annotation_layer_copy fails bc the annotation zip is empty
```

## Your Environment for bug

- Current backend
- wklibs version 0.9.4
- [ ] Specific to long-running jobs (set `jobsEnabled=true` in `application.conf`)
- [ ] Specific to webKnossos.org (set `isDemoInstance=true` in `application.conf`)

**Comment by fm3** [comment]
We decided to include header.wkw files in empty volume annotation downloads, as part of #6128. This should solve one of the problems mentioned here. 

However, if there are _no_ volume layers, I’m not sure if switching from pure nml to zip would be a good idea. This might break some existing client code, and creates performance load due to the zipping operation. I’d say this should probably rather be solved in the python client. Do you agree @philippotto @jstriebel ?

**Comment by jstriebel** [comment]
> However, if there are _no_ volume layers, I’m not sure if switching from pure nml to zip would be a good idea. This might break some existing client code, and creates performance load due to the zipping operation. I’d say this should probably rather be solved in the python client. Do you agree @philippotto @jstriebel ?

Agreed, I think it should still be the pure NML, to keep compatibility

**Comment by fm3** [comment]
Thanks for your input! Then I’d say the parts of this issue that we want to implement, have been implemented in the meantime as part of #6128 so I’m closing this
